### ISSUE TITLE
Commercial product leakage to package.json

### ISSUE THREAD

**Issue by knoobie** [vaadin/flow#13749]
### Description of the bug

Currently commercial projects like `vaadin-grid-pro` are added to the package.json and package-lock.json even tho only `vaadin-core` is used as dependency.

### Expected behavior

Not used dependencies / commercial products are added into the package.json or package-lock.json.

### Minimal reproducible example

- Create a V23 app in start.vaadin.com
- Replace `vaadin` dependency with `vaadin-core`
- Delete package.json & package-lock.json 
- mvn
- Look at the nearly created package.json containing commercial products 

### Versions

- Vaadin / Flow version: V23 + V23.1.0.beta1 tested
- Java version: 11

**Comment by Artur-** [comment]
Yes, there should be separate version jsons in the two jars. Earlier it did not matter that vaadin-core listed all the versions as only what was found by scanning for `@NpmPackage` was installed but now I think Flow installs everything it finds in versions.json.

Technically it might be easier to name the files differently in the two jars so you can just fetch from the classpath using a unique name

**Comment by haijian-ccc** [comment]
This is actually a critical bug. It's not uncommon that companies have license check in the build script, upon detecting commercial licenses, the build fails which makes Vaadin 23 not usable. And there is no workaround, as Vaadin always forces the commercial products, even the user only intends to use the free ones with vaadin-core.

**Comment by ZheSun88** [comment]
I have created this [PR](https://github.com/vaadin/platform/pull/3060) to generate different json file for vaadin and vaadin-core

and the json will be included into different jars. 

```
Platform includes free components into vaadin-versions-core.json and includes it into vaadin-core JAR
Platform includes commertial components into vaadin-versions.json and includes it into vaadin JAR
```
i think flow will need to do corresponding changes to make it include the correct json into the generated package.json files.
